src/logger/POSTRequestLogger.ts

```diff
@@ -0,0 +1,127 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { Logger, LogLevel } from '../index';
+import IntervalScheduler from '../scheduler/IntervalScheduler';
+
+export default class POSTRequestLogger implements Logger {
+  private logCapture: string[] = [];
+  private logSequenceNumber: number = 0;
+  private lock = false;
+  private static BATCH_SIZES = 75;
+  private intervalScheduler: IntervalScheduler;
+
+  constructor(
+    private name: string,
+    private meetingId: string,
+    private attendeeId: string,
+    private intervalMs: number,
+    private baseURL: string,
+    private level = LogLevel.WARN,
+  ) {
+    this.intervalScheduler = new IntervalScheduler(this.intervalMs);
+    this.publishLogs(POSTRequestLogger.BATCH_SIZES);
+    window.addEventListener('unload', () => {
+      this.intervalScheduler.stop();
+      if(!navigator.sendBeacon){
+        return
+      }
+      const body = JSON.stringify({
+        meetingId: this.meetingId,
+        attendeeId: this.attendeeId,
+        appName: this.name,
+        logs: this.logCapture
+      });
+      navigator.sendBeacon(this.baseURL, body);
+    });
+  }
+
+  debug(debugFunction: () => string): void {
+    if (LogLevel.DEBUG < this.level) {
+      return;
+    }
+    this.log(LogLevel.DEBUG, debugFunction());
+  }
+
+  info(msg: string): void {
+    this.log(LogLevel.INFO, msg);
+  }
+
+  warn(msg: string): void {
+    this.log(LogLevel.WARN, msg);
+  }
+
+  error(msg: string): void {
+    this.log(LogLevel.ERROR, msg);
+  }
+
+  setLogLevel(level: LogLevel): void {
+    this.level = level;
+  }
+
+  getLogLevel(): LogLevel {
+    return this.level;
+  }
+
+  publishLogs(batchSize: number): void {
+    this.intervalScheduler.start(async () => {
+      if (this.lock === true || this.logCapture.length === 0) {
+        return;
+      }
+      this.lock = true;
+      const batch = this.logCapture.slice(0, batchSize);
+      const body = JSON.stringify({
+        meetingId: this.meetingId,
+        attendeeId: this.attendeeId,
+        appName: this.name,
+        logs: batch,
+      });
+      try {
+        const response = await fetch(this.baseURL, {
+          method: 'POST',
+          body,
+        });
+        if (response.status === 200) {
+          this.logCapture = this.logCapture.slice(batch.length);
+        }
+      } catch (error) {
+        this.warn('[POSTRequestLogger] ' + error.message);
```

Comment:
Suggest adding an explicit `console.warn`. If the endpoint behind POSTRequestLogger is completely down, it will be difficult to detect since the warn log about that issue will never make it. (See note below about removing console logging for regular messages).